src/locale.rs

```diff
@@ -10,18 +10,20 @@ const CHAR_MAX: libc::c_char = libc::c_char::MAX;
 
 /// Return the first character of a C string, or None if null, empty, has a length more than 1, or negative.
 unsafe fn first_char(s: *const libc::c_char) -> Option<char> {
-    #[allow(unused_comparisons, clippy::absurd_extreme_comparisons)]
-    if !s.is_null() && *s > 0 && *s <= 127 && *s.offset(1) == 0 {
-        Some((*s as u8) as char)
-    } else {
-        None
+    unsafe {
+        #[allow(unused_comparisons, clippy::absurd_extreme_comparisons)]
+        if !s.is_null() && *s > 0 && *s <= 127 && *s.offset(1) == 0 {
+            Some((*s as u8) as char)
+        } else {
+            None
+        }
     }
 }
 
 /// Convert a libc lconv to a Locale.
 unsafe fn lconv_to_locale(lconv: &libc::lconv) -> Locale {
-    let decimal_point = first_char(lconv.decimal_point).unwrap_or('.');
```

Comment:
re a7d4d521a80 (Make unsafes more granular, 2025-10-23),
yeah better squash this as well, don't make reviewers read poorly-styled `cargo fix` output